src/vm/dllimport.cpp

```diff
@@ -5747,25 +5755,34 @@ class LoadLibErrorTracker
                 priority = const_priorityCouldNotLoad;
                 break;
         }
-
-        UpdateHR(priority, HRESULT_FROM_WIN32(dwLastError));
+        UpdateHRWin(priority, HRESULT_FROM_WIN32(dwLastError));
+#endif
     }
 
     // Sets the error code to HRESULT as could not load DLL
     void TrackHR_CouldNotLoad(HRESULT hr)
     {
-        UpdateHR(const_priorityCouldNotLoad, hr);
+        UpdateHRWin(const_priorityCouldNotLoad, hr);
     }
     
     HRESULT GetHR()
     {
         return m_hr;
     }
 
+    SString GetMessage()
+    {
+        return m_message;
+    }
+
     void DECLSPEC_NORETURN Throw(SString &libraryNameOrPath)
     {
         STANDARD_VM_CONTRACT;
 
+#ifdef FEATURE_PAL
+        SString hrString = (SString) GetMessage();
```

Comment:
No need to store it in a local. You can just pass it into COMPlusThrow directly.